### PR TITLE
Zappy 48 map

### PR DESCRIPTION
**Pull request by PaulArbez**

### CODE REVIEW

**Review by lmaoleo — APPROVED** [review]

File: client/gui/src/Gui.cpp [hunk 1]
```diff
@@ -21,11 +24,23 @@ void Gui::start()
...
-        DrawText("Congrats! You created your first window!", 190, 200, 20, LIGHTGRAY);
-        EndDrawing();
+        if(IsKeyDown(KEY_W))
+            _camera.moveCamera(_camera.FORWARD);
+        if(IsKeyDown(KEY_S))
+            _camera.moveCamera(_camera.BACKWARD);
+        if(IsKeyDown(KEY_A))
+            _camera.moveCamera(_camera.LEFT);
+        if(IsKeyDown(KEY_D))
+            _camera.moveCamera(_camera.RIGHT);
```
Comment (Kapsulon) [inline client/gui/src/Gui.cpp]: put this into the camera class

File: client/gui/src/Map.cpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+void Map::init_map()
+{
+    bool color = false;
+
+    for (int i = 0; i < _x; i++) {
+        std::vector<MyCube> tmp;
+        for (int j = 0; j < _y; j++) {
+            MyCube cube;
+            cube.set_position(Vector3{(float)i - (_x / 2.0f - 0.5f), 0, (float)j - (_y / 2.0f - 0.5f)});
```
Comment (Kapsulon) [inline client/gui/src/Map.cpp]: use raylib namespace

File: client/gui/src/MyCube.hpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+class MyCube {
+    public:
+        MyCube();
+        ~MyCube();
+        void draw();
+        Vector3 get_position() { return _position; }
+        void set_position(Vector3 position) { _position = position; }
+        void set_color(raylib::Color color) { _color = color; }
+        void set_border_color(Color color) { _border_color = color; }
```
Comment (Kapsulon) [inline client/gui/src/MyCube.hpp]: usePascalCase and raylib namespace

File: client/gui/src/Map.hpp [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+#define MAP_HPP_
+
+#include "MyCube.hpp"
+
+class Map {
+    public:
+        Map();
+        Map(int x, int y);
+        ~Map();
+        void init_map();
```
Comment (Kapsulon) [inline client/gui/src/Map.hpp]: usePascalCase

File: client/gui/src/items/Item.cpp [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+{
+    _position.SetX(x);
+    _position.SetY(y);
+    _position.SetZ(z);
+    _quantity = quantity;
+}
+
+Item::~Item()
+{
+}
```
Comment (Kapsulon) [inline client/gui/src/items/Item.cpp]: put destructor to default and remove these lines

**Review by Kapsulon — CHANGES_REQUESTED** [review]

**Review by Kapsulon — APPROVED** [review]
gud

**Pull request merged**